packages/graphql-transformers-e2e-tests/src/cognitoUtils.ts

```diff
@@ -163,3 +176,212 @@ export async function createUserPoolClient(client: CognitoClient, userPoolId: st
+                PolicyDocument: {
+                    Version: '2012-10-17',
+                    Statement: [
+                    {
+                        Effect: 'Allow',
+                        Action: [
+                            'appsync:GraphQL'
+                        ],
+                        Resource: [{
+                            'Fn::Join': [
+                                '',
+                                [
+                                    'arn:aws:appsync:',
+                                    { Ref: 'AWS::Region' },
+                                    ':',
+                                    { Ref: 'AWS::AccountId' },
+                                    ':apis/',
+                                    {
+                                        'Fn::GetAtt': [
+                                            'GraphQLAPI',
+                                            'ApiId'
+                                        ]
+                                    },
+                                    '/*',
+                                ],
+                            ],
+                        }],
+                    }],
+                },
+            }),
+        ]
+    });
+
+    const identityPool = new cfnCognito.IdentityPool({
+        IdentityPoolName: IDENTITY_POOL_NAME,
+        CognitoIdentityProviders: [
+            {
+                ClientId: {
+                    Ref: 'UserPoolClient'
+                },
+                ProviderName: {
+                    'Fn::Sub': [
+                        'cognito-idp.${region}.amazonaws.com/${client}',
+                        {
+                            'region': {
+                                Ref: 'AWS::Region'
+                            },
+                            'client': USER_POOL_ID
+                        }
+                    ]
+                }
+            } as unknown,
+            {
+                ClientId: {
+                    Ref: 'UserPoolClientWeb'
+                },
+                ProviderName: {
+                    'Fn::Sub': [
+                        'cognito-idp.${region}.amazonaws.com/${client}',
+                        {
+                            'region': {
+                                Ref: 'AWS::Region'
+                            },
+                            'client': USER_POOL_ID
+                        }
+                    ]
+                }
+            } as unknown,
+        ],
+        AllowUnauthenticatedIdentities: true
+    });
+
+    const identityPoolRoleMap = new cfnCognito.IdentityPoolRoleAttachment({
+        IdentityPoolId: { Ref: 'IdentityPool' } as unknown as string,
+        Roles: {
+            'unauthenticated': { 'Fn::GetAtt': [ 'UnauthRole', 'Arn' ] },
+            'authenticated': { 'Fn::GetAtt': [ 'AuthRole', 'Arn' ] }
+        }
+    });
+
+    const userPoolClientWeb = new cfnCognito.UserPoolClient({
+        ClientName: USER_POOL_CLIENTWEB_NAME,
```

Comment:
Why don't we generate a secret for userPoolClientWeb?

Reply:
I don't think this is needed as the clientID is this e2e is testing auth resolver logic against Elasticsearch responses.